gnovm/pkg/gnolang/store.go

```diff
@@ -689,6 +759,17 @@ func (ds *defaultStore) IterMemPackage() <-chan *std.MemPackage {
 	}
 }
 
+func (ds *defaultStore) SetGasMeter(gm store.GasMeter) {
+	ds.gasMeter = gm
+}
```

Comment:
I think this ought to go in `BeginTransaction`, and the store shouldn't use a gas store otherwise.

Reply:
It makes sense to count gas for the store independently since future features will dissociate CPU gas from storage gas, allowing costs to cover `cpuCost + storageCost + futureStorageRent`. Additionally, realms could have different storage cost ratios—some sponsored by the chain, others transitioned to 'archives,' or even storage classes with varying configurations based on price tiers.

Reply:
> I think this ought to go in `BeginTransaction`, and the store shouldn't use a gas store otherwise.
@thehowl Agreed, I will assign the gasMeter in BeginTransaction and remove this method.